fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1685,6 +1658,24 @@ private Bundle processBundleEntries(Bundle requestBundle, Map<Integer, Entry> va
                         "HEAD" + requestEntriesByMethod.get(HTTPVerb.Value.HEAD));
             }
 
+            // If we're working on a 'transaction' type interaction, or an interaction
+            // where we're processing only GET requests, then start a new transaction now.
+            BundleType.Value bundleType = requestBundle.getType().getValueAsEnum();
+            if (bundleType == BundleType.Value.TRANSACTION ||
+                    (!requestEntriesByMethod.get(HTTPVerb.Value.GET).isEmpty() &&
+                            requestEntriesByMethod.get(HTTPVerb.Value.DELETE).isEmpty() &&
+                            requestEntriesByMethod.get(HTTPVerb.Value.POST).isEmpty() &&
+                            requestEntriesByMethod.get(HTTPVerb.Value.PUT).isEmpty() &&
+                            requestEntriesByMethod.get(HTTPVerb.Value.PATCH).isEmpty() &&
+                            requestEntriesByMethod.get(HTTPVerb.Value.HEAD).isEmpty())) {
```

Comment:
I'm really not sure where we stand in terms of HEAD requests from within a bundle, but technically that should be safe to do within the same transaction as the GETs.  Would it have any benefit?

Reply:
Makes sense to me to treat GET and HEAD the same - have updated the code to do so.